components/cdc/src/delegate.rs

```diff
@@ -608,6 +654,48 @@ impl Delegate {
     }
 
     fn sink_put(
+        &mut self,
+        mut put: PutRequest,
+        is_one_pc: bool,
+        txn_rows: &mut HashMap<Vec<u8>, EventRow>,
+        raw_rows: &mut HashMap<Vec<u8>, EventRow>,
+        read_old_value: impl FnMut(&mut EventRow, TimeStamp) -> Result<()>,
+    ) -> Result<()> {
+        let key = put.take_key();
+        let key_mode = APIV2::parse_key_mode(&key);
+
+        if key_mode == KeyMode::Raw {
+            self.sink_raw_put(put, raw_rows)
+        } else {
+            self.sink_txn_put(put, is_one_pc, txn_rows, read_old_value)
```

Comment:
If `kv_api` is `rawkv`, we don't need to `sink_txn_put()`, and vice versa. To be better, we can set appropriate scan range in `Initializer`.